docs/userguide/async.md

```diff
@@ -0,0 +1,74 @@
+# Asynchronous Workflows
```

Comment:
Note: Since there are several comments by @AdrianSosic already regarding the user guide, I only considered his comments for now and will give the user guide a fresh review once those comments have been addressed. Ping me here once this has happened.